psi4/src/psi4/fisapt/fisapt.cc

```diff
@@ -2768,56 +2717,40 @@ void FISAPT::felst()
 
     // => a <-> b <= //
 
-    std::shared_ptr<BasisSet> jkfit = reference_->get_basisset("DF_BASIS_SCF");
-    size_t nQ = jkfit->nbf();
-
-    std::shared_ptr<DFERI> df = DFERI::build(primary_,jkfit,options_);
-    df->clear();
-
-    std::vector<std::shared_ptr<Matrix> > Cs;
-    Cs.push_back(matrices_["Locc0A"]);
-    Cs.push_back(matrices_["Locc0B"]);
-    std::shared_ptr<Matrix> Call = Matrix::horzcat(Cs);
-    Cs.clear();
-
-    df->set_C(Call);
-    df->set_memory(doubles_);
-
-    size_t offset = 0;
-    df->add_space("a",offset,offset+na); offset += na;
-    df->add_space("b",offset,offset+nb); offset += nb;
-
-    df->add_pair_space("Aaa", "a", "a");
-    df->add_pair_space("Abb", "b", "b");
+    int nT = 1;
+    #ifdef _OPENMP
+        nT = Process::environment.get_n_threads();
+    #endif
+    
+    // => Get integrals from DF_Helper <= //
+    std::shared_ptr<df_helper::DF_Helper> dfh = std::shared_ptr<df_helper::DF_Helper>(new 
+        df_helper::DF_Helper(primary_, reference_->get_basisset("DF_BASIS_SCF")));
+    dfh->set_memory(doubles_);
+    dfh->set_method("DIRECT");
+    dfh->set_nthreads(nT);
+    dfh->initialize(); 
+    dfh->print_header();    
 
-    df->print_header();
-    df->compute();
+    dfh->add_space("a", matrices_["Locc0A"]);
+    dfh->add_space("b", matrices_["Locc0B"]);
 
-    std::map<std::string, std::shared_ptr<Tensor> >& ints = df->ints();
-    std::shared_ptr<Tensor> AaaT = ints["Aaa"];
-    std::shared_ptr<Tensor> AbbT = ints["Abb"];
+    dfh->add_transformation("Aaa", "a", "a", "pqQ");
+    dfh->add_transformation("Abb", "b", "b", "pqQ");
 
-    df.reset();
+    dfh->transform();
 
+    size_t nQ = dfh->get_naux();
     std::shared_ptr<Matrix> QaC(new Matrix("QaC", na, nQ));
     double** QaCp = QaC->pointer();
-    FILE* Aaaf = AaaT->file_pointer();
-    fseek(Aaaf,0L,SEEK_SET);
     for (size_t a = 0; a < na; a++) {
-        fseek(Aaaf,(a * na + a) * nQ * sizeof(double), SEEK_SET);
-        size_t statusvalue=fread(QaCp[a], sizeof(double), nQ, Aaaf);
+        dfh->fill_tensor("Aaa", QaCp[a], std::make_pair(a, a + 1), std::make_pair(a, a + 1));
     }
-    AaaT.reset();
 
     std::shared_ptr<Matrix> QbC(new Matrix("QbC", nb, nQ));
     double** QbCp = QbC->pointer();
-    FILE* Abbf = AbbT->file_pointer();
-    fseek(Abbf,0L,SEEK_SET);
     for (size_t b = 0; b < nb; b++) {
-        fseek(Abbf,(b * nb + b) * nQ * sizeof(double), SEEK_SET);
-        size_t statusvalue=fread(QbCp[b], sizeof(double), nQ, Abbf);
+        dfh->fill_tensor("Abb", QbCp[b], std::make_pair(b, b + 1), std::make_pair(b, b + 1));
```

Comment:
For this signature do you really want it as a `std::pair`? In C++17 we can skip the `make_pair`, but for now if we used `std::vectors` we could do: ` dfh->fill_tensor("Abb", QbCp[b], {b, b + 1}, {b, b + 1});`. You would have to add one more check ensuring that you are only giving it two elements. Just a comment on the syntactic sugar.